### PR TITLE
add sign up route,controller,service and utils

### PR DESCRIPTION
**Pull request by didasmbalanya**
#### What does this PR do?
Adds the create user feature.
#### Description of Task to be completed?
Enable new users to sign up.
#### How should this be manually tested?
Post Man
#### What are the relevant pivotal tracker stories?
User can sign up #167972378
Screen Shot 2019-08-22 at 18 37 27

#### Questions:

### CODE REVIEW

File: src/controllers/userController.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    if (foundUser) {
+      res.status(409).send({ error: `User with email ${req.body.email} already exists` });
+    } else {
+      try {
+        req.body.is_admin = false;
+        if (req.body.password === req.body.confirm_password) {
+          req.body.password = bcrypt.hashSync(req.body.password, 8);
+          const newUser = await userService.addUser(req.body);
+          const token = jwt.sign({ email: req.body.email }, secret, { expiresIn: '3h' });
+          res.status(201).send({ data: getPublicProfile(newUser), token });
```
Comment (mucyomiller) [inline src/controllers/userController.js]: Great work, however, you forgot to put status code also in the response payload 🤔

**Review by mucyomiller — APPROVED** [review]
LGTM

**Review by amilykassim — APPROVED** [review]

**Review by gik-hub — CHANGES_REQUESTED** [review]
Looks good to me.
You just need to add a response code to the response

**Review by mucyomiller — APPROVED** [review]
Now ok 👍

**Review by gik-hub — APPROVED** [review]

**Pull request merged**